### PR TITLE
Update Roslyn to 4.4.0-2.final (UT project)

### PR DESCRIPTION
**Pull request by costin-zaharia-sonarsource**

### CODE REVIEW

**Review by csaba-sagi-sonarsource — APPROVED** [review]
LGTM!

File: analyzers/tests/SonarAnalyzer.UnitTest/Helpers/OperationExecutionOrderTest.cs [hunk 1]
```diff
@@ -52,7 +52,7 @@ public void Linear(bool reverseOrder)
                 "LocalReference: a",
                 "LocalReference: b",
                 "Literal: 1",
-                "Binary: b + 1",
+                "BinaryOperator: b + 1",
```
Comment (costin-zaharia-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/Helpers/OperationExecutionOrderTest.cs:55]: It looks like they've changed this back to `BinaryOperator`.

File: analyzers/tests/SonarAnalyzer.UnitTest/Rules/EncryptionAlgorithmsShouldBeSecureTest.cs [hunk 2]
```diff
@@ -26,29 +26,47 @@ namespace SonarAnalyzer.UnitTest.Rules
...
+#else
+
+        [TestMethod]
+        public void EncryptionAlgorithmsShouldBeSecure_VB_NetStandard21_Net48() =>
+            builderVB.AddPaths(@"EncryptionAlgorithmsShouldBeSecure_NetStandard21.Net48.vb")
+                     .WithOptions(ParseOptionsHelper.VisualBasicLatest)
+                     .AddReferences(MetadataReferenceFacade.NETStandard21.Concat(GetAdditionalReferences()))
+                     .Verify();
+
+#endif
```
Comment (costin-zaharia-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/Rules/EncryptionAlgorithmsShouldBeSecureTest.cs:68]: I've extracted 2 separate files with differences, to make them obvious.

File: analyzers/tests/SonarAnalyzer.UnitTest/SonarAnalyzer.UnitTest.csproj [hunk 3]
```diff
@@ -35,9 +35,9 @@
     <PackageReference Include="Moq" Version="4.18.2" />
     <PackageReference Include="MSTest.TestAdapter" Version="2.2.10" />
     <PackageReference Include="MSTest.TestFramework" Version="2.2.10" />
-    <PackageReference Include="Microsoft.CodeAnalysis.Workspaces.Common" Version="4.2.0" />
-    <PackageReference Include="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="4.2.0" />
-    <PackageReference Include="Microsoft.CodeAnalysis.VisualBasic.Workspaces" Version="4.2.0" />
+    <PackageReference Include="Microsoft.CodeAnalysis.Workspaces.Common" Version="4.4.0-2.final" />
+    <PackageReference Include="Microsoft.CodeAnalysis.CSharp.Workspaces" Version="4.4.0-2.final" />
+    <PackageReference Include="Microsoft.CodeAnalysis.VisualBasic.Workspaces" Version="4.4.0-2.final" />
```
Comment (costin-zaharia-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/SonarAnalyzer.UnitTest.csproj:40]: The latest at this moment.

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ClassShouldNotBeAbstract.cs [hunk 4]
```diff
@@ -108,6 +108,6 @@ public abstract class View3Derived : SomeUnknownType // Noncompliant
                                                          // Error@-1 [CS0246]
     {
         public string Content3 { get; }
-        public override int Content1 { get { return 1; } } // Error [CS0115]
```
Comment (costin-zaharia-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/ClassShouldNotBeAbstract.cs:111]: Error [CS0115] is no longer raised.

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/EncryptionAlgorithmsShouldBeSecure_NetStandard21.Net48.vb [hunk 5]
```diff
@@ -0,0 +1,24 @@
+﻿Imports System
+Imports System.Security.Cryptography
+
+Namespace Tests.Diagnostics
+    Class RSAEncryptionPaddingTest
+        Private rsaProvider As RSACryptoServiceProvider = New RSACryptoServiceProvider()
+
+        Private Sub InvocationSetsAllowedValue(ByVal data As Byte(), ByVal padding As RSAEncryptionPadding, ByVal genericRSA As RSA)
+            Dim bytesWritten = Nothing
+            rsaProvider.TryEncrypt(data, Nothing, RSAEncryptionPadding.OaepSHA256, bytesWritten)
```
Comment (costin-zaharia-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/EncryptionAlgorithmsShouldBeSecure_NetStandard21.Net48.vb:10]: net 48 can use `ReadOnlySpan<Byte>`

File: analyzers/tests/SonarAnalyzer.UnitTest/TestCases/EncryptionAlgorithmsShouldBeSecure_NetStandard21.Net7.vb [hunk 6]
```diff
@@ -0,0 +1,24 @@
+﻿Imports System
+Imports System.Security.Cryptography
+
+Namespace Tests.Diagnostics
+    Class RSAEncryptionPaddingTest
+        Private rsaProvider As RSACryptoServiceProvider = New RSACryptoServiceProvider()
+
+        Private Sub InvocationSetsAllowedValue(ByVal data As Byte(), ByVal padding As RSAEncryptionPadding, ByVal genericRSA As RSA)
+            Dim bytesWritten = Nothing
+            rsaProvider.TryEncrypt(data, Nothing, RSAEncryptionPadding.OaepSHA256, bytesWritten) ' Error [BC37319]
```
Comment (costin-zaharia-sonarsource) [inline analyzers/tests/SonarAnalyzer.UnitTest/TestCases/EncryptionAlgorithmsShouldBeSecure_NetStandard21.Net7.vb:10]: On Net 7, Visual Basic cannot use  `ReadOnlySpan<Byte>`

**Review by csaba-sagi-sonarsource — APPROVED** [review]
LGTM!

**Pull request merged**